keystore/keyloader/hashicorp/vault_cli.go

```diff
@@ -56,51 +68,106 @@ func ParseCLIParametersFromFlags(flags *flag.FlagSet, prefix string) *VaultCLIOp
+		}
+		options.EnableTLS = val
+	}
+
+	namerFunc := network.ClientNameConstructorFunc()
+	if f := flags.Lookup(namerFunc("vault", "ca", "")); f != nil {
+		newCAPathFlagValue := f.Value.String()
+		if options.CAPath != "" && newCAPathFlagValue != "" {
+			log.Errorf("Flags `%s` (deprecated) and `%s` cant be provided simultaneously", "vault_tls_ca_path", f.Name)
+			os.Exit(1)
+		}
+
+		if newCAPathFlagValue != "" {
+			options.CAPath = f.Value.String()
+		}
+	}
+	if f := flags.Lookup(namerFunc("vault", "sni", "")); f != nil {
+		options.SNI = f.Value.String()
+	}
+	if f := flags.Lookup(namerFunc("vault", "cert", "")); f != nil {
 		options.ClientCert = f.Value.String()
 	}
-	if f := flags.Lookup(prefix + "vault_tls_client_key"); f != nil {
+	if f := flags.Lookup(namerFunc("vault", "key", "")); f != nil {
 		options.ClientKey = f.Value.String()
 	}
-	if f := flags.Lookup(prefix + "vault_tls_transport_enable"); f != nil {
-		val, err := strconv.ParseBool(f.Value.String())
+	if f := flags.Lookup(namerFunc("vault", "auth", "")); f != nil {
+		v, err := strconv.ParseInt(f.Value.String(), 10, 64)
 		if err != nil {
-			log.WithField("value", f.Value.String()).Fatalf("Can't cast %s to bool value", prefix+"vault_tls_transport_enable")
+			log.WithField("value", f.Value.String).Fatalf("Can't cast %s to integer value", f.Name)
 		}
-		options.EnableTLS = val
+		options.Auth = tls.ClientAuthType(v)
+	}
+
+	var err error
+	ocspConfig, err := network.NewOCSPConfigByName(flags, "vault", namerFunc)
+	if err != nil {
+		log.WithError(err).Fatalf("Can't parse Vault OCSPConfig")
+	}
+	crlConfig, err := network.NewCRLConfigByName(flags, "vault", namerFunc)
+	if err != nil {
+		log.WithError(err).Fatalf("Can't parse Vault CRLConfig")
+	}
+	options.CertVerifier, err = network.NewCertVerifierFromConfigs(ocspConfig, crlConfig)
+	if err != nil {
+		log.WithError(err).Fatalf("Can't parse Vault CertVerifier")
 	}
 
 	return &options
 }
 
-// TLSConfig return TLS configuration needed to connect to HashiCorp Vault
-func (options *VaultCLIOptions) TLSConfig() *api.TLSConfig {
-	return &api.TLSConfig{
-		ClientKey:  options.ClientKey,
-		ClientCert: options.ClientCert,
-		CAPath:     options.CAPath,
+// VaultHTTPClient returns api.Config connection configuration
+func (options *VaultCLIOptions) VaultHTTPClient() (*http.Client, error) {
+	transport := cleanhttp.DefaultPooledTransport()
+	client := &http.Client{
+		Transport: transport,
 	}
+
+	vaultURL, err := url.ParseRequestURI(options.Address)
+	if err != nil {
+		log.WithError(err).WithField("address", options.Address).Errorln("Invalid Vault address provided")
+		return nil, err
+	}
+
+	if options.EnableTLS {
+		serverName := network.SNIOrHostname(options.SNI, vaultURL.Host)
+		tlsConfig, err := network.NewTLSConfig(serverName, options.CAPath, options.ClientKey, options.ClientCert, options.Auth, options.CertVerifier)
+		if err != nil {
+			return nil, err
+		}
+
+		transport.TLSHandshakeTimeout = 10 * time.Second
```

Comment:
is there any reason to use exactly 10 seconds or we can use our common value https://github.com/cossacklabs/acra/blob/master/network/deadline_listener.go#L27 1 minute?

Reply:
I digged into hashicorp lib and found that time, decided to make the same.

https://github.com/hashicorp/vault/blob/main/api/client.go#L230

What do you think is better?

Reply:
oh, I didn't note that it is for TLSHandshake and they use 60sec for timeout too... okay, lets leave 10sec for handshake.